pkg/iscsi/iscsi_util.go

```diff
@@ -517,51 +544,70 @@ func (util *ISCSIUtil) DetachDisk(
 	for _, portal := range portals {
-		logoutArgs := []string{
-			"-m", "node", "-p", portal, "-T", iqn, "--logout",
-		}
-		deleteArgs := []string{
-			"-m", "node", "-p", portal, "-T", iqn, "-o", "delete",
-		}
+		logoutArgs := []string{"-m", "node", "-p", portal, "-T", iqn, "--logout"}
+		deleteArgs := []string{"-m", "node", "-p", portal, "-T", iqn, "-o", "delete"}
 		if found {
 			logoutArgs = append(logoutArgs, []string{"-I", iface}...)
 			deleteArgs = append(deleteArgs, []string{"-I", iface}...)
 		}
-		glog.Infof(
-			"iscsi: log out target %s iqn %s iface %s",
-			portal, iqn, iface,
-		)
-		out, err := c.exec.Command("iscsiadm", logoutArgs...).CombinedOutput()
+		logrus.Infof("iscsi: log out target %s iqn %s iface %s", portal, iqn, iface)
+		out, err := exec.Command("iscsiadm", logoutArgs...).CombinedOutput()
+		err = ignoreExitCodes(err, exit_ISCSI_ERR_NO_OBJS_FOUND, exit_ISCSI_ERR_SESS_NOT_FOUND)
 		if err != nil {
-			glog.Errorf("iscsi: failed to detach disk Error: %s", string(out))
+			logrus.Errorf("iscsi: failed to detach disk Error: %s", string(out))
+			return err
 		}
 		// Delete the node record
-		glog.Infof("iscsi: delete node record target %s iqn %s", portal, iqn)
-		out, err = c.exec.Command("iscsiadm", deleteArgs...).CombinedOutput()
+		logrus.Infof("iscsi: delete node record target %s iqn %s", portal, iqn)
+		out, err = exec.Command("iscsiadm", deleteArgs...).CombinedOutput()
+		err = ignoreExitCodes(err, exit_ISCSI_ERR_NO_OBJS_FOUND, exit_ISCSI_ERR_SESS_NOT_FOUND)
 		if err != nil {
-			glog.Errorf(
-				"iscsi: failed to delete node record Error: %s",
-				string(out),
-			)
+			logrus.Errorf("iscsi: failed to delete node record Error: %s", string(out))
+			return err
 		}
 	}
 	// Delete the iface after all sessions have logged out
 	// If the iface is not created via iscsi plugin, skip to delete
 	if initiatorName != "" && found && iface == (portals[0]+":"+volName) {
 		deleteArgs := []string{"-m", "iface", "-I", iface, "-o", "delete"}
-		out, err := c.exec.Command("iscsiadm", deleteArgs...).CombinedOutput()
+		out, err := exec.Command("iscsiadm", deleteArgs...).CombinedOutput()
+		err = ignoreExitCodes(err, exit_ISCSI_ERR_NO_OBJS_FOUND, exit_ISCSI_ERR_SESS_NOT_FOUND)
 		if err != nil {
-			glog.Errorf("iscsi: failed to delete iface Error: %s", string(out))
+			logrus.Errorf("iscsi: failed to delete iface Error: %s", string(out))
+			return err
 		}
 	}
 
-	if err := os.RemoveAll(targetPath); err != nil {
-		glog.Errorf("iscsi: failed to remove mount path Error: %v", err)
+	return nil
+}
+
+func ignoreExitCodes(err error, ignoredExitCodes ...int) error {
+	exitError, ok := err.(utilexec.ExitError)
+	if !ok {
 		return err
 	}
-
-	return nil
+	for _, code := range ignoredExitCodes {
+		if exitError.ExitStatus() == code {
+			logrus.Debugf("ignored iscsiadm exit code %d", code)
```

Comment:
Should we need to change into info?

Reply:
Thought was to have as debug log .. anyways changed it to infof as it will logged in rare cases